pkg/cluster/workloadidentityresources.go

```diff
@@ -124,3 +130,37 @@ func (m *manager) generateAuthenticationConfig() (*configv1.Authentication, erro
 		},
 	}, nil
 }
+
+// getPlatformWorkloadIdentityFederatedCredName returns a federated identity credential name.
+// name length 3-120 characters. It must be alphanumeric, dash, underscore.
+func (m *manager) getPlatformWorkloadIdentityFederatedCredName(sa string, identity api.PlatformWorkloadIdentity) (string, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return "", fmt.Errorf("getPlatformWorkloadIdentityFederatedCredName called for a CSP cluster")
+	}
+
+	if strings.TrimSpace(sa) == "" {
+		return "", fmt.Errorf("service account name is required")
+	}
+
+	identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+	if err != nil {
+		return "", err
+	}
+	clusterResourceId, err := azure.ParseResourceID(m.doc.OpenShiftCluster.ID)
+	if err != nil {
+		return "", err
+	}
+
+	clusterResourceKey := fmt.Sprintf("%s-%s-%s", clusterResourceId.SubscriptionID, clusterResourceId.ResourceGroup, clusterResourceId.ResourceName)
```

Comment:
Why do we need this complicated calculation?

If you want to be safe, just do:
- hash the SA name w/ SHA1
- hash the identity resource URI w/ SHA1
- return `sa-hash`-`resourceURI-hash`

This way we can perform additional service account federations if needed in the future.

Re collisions: https://stackoverflow.com/questions/4014090/is-it-safe-to-ignore-the-possibility-of-sha-collisions-in-practice